### PR TITLE
Add `partition` to Data.Array

### PR DESCRIPTION
**Pull request by raichoo**
I needed a `partition` function for Arrays so I made one. Pointers welcome, this currently feels like the straightforward way to implement it.

### CODE REVIEW

**Comment by raichoo** [comment]
Interestingly enough this seems to break in a line that I didn't touch. I'm not familiar with the tools complaining here so I'm not sure if this is actually my mistake, it certainly looks like it isn't to me.

**Comment by garyb** [comment]
Ugh, it's not you, jshint broke something a while ago. Can you change [this line](https://github.com/purescript/purescript-arrays/blob/f2ed8d23b1c9be70370698178e4bcf447c79bcc4/.jshintrc#L8) to `"strict": "global",` please? That should fix it.

I'm also amazed we don't already have `partition` :smile:

**Comment by raichoo** [comment]
Good to know :) So was I ^^.

**Comment by raichoo** [comment]
All fixed now.

File: src/Data/Array.purs [hunk 1]
```diff
@@ -369,6 +370,13 @@ concatMap = flip bind
 -- | creating a new array.
 foreign import filter :: forall a. (a -> Boolean) -> Array a -> Array a
 
+-- | Partition an array using a predicate function, creating a set of
+-- | new arrays. One for the values satisfying the predicate function
+-- | and one for values that don't.
+foreign import partition :: forall a. (a -> Boolean)
+                         -> Array a
+                         -> { yes :: Array a, no :: Array a }
```
Comment (paf31) [inline src/Data/Array.purs]: I like the return type, I frequently have to consult Hackage to remember the right way round in Haskell.

**Comment by paf31** [comment]
:+1: LGTM, but could you please run `pulp docs`?

**Comment by raichoo** [comment]
On its way. The output surprised me however. I looks like it touched more of the documentation than I actually changed.

**Comment by paf31** [comment]
It looks like we forgot to build it last time something was merged.

Looks good, thanks!

**Comment by sharkdp** [comment]
Looks good to me, too.

**Comment by garyb** [comment]
Thanks @raichoo!

**Pull request merged**